x-pack/legacy/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -526,22 +619,22 @@ export class DataRecognizer {
       })
     );
     // merge all results
-    return [].concat(...results);
+    return ([] as ObjectExistResponse[]).concat(...results);
   }
 
   // find all existing savedObjects for a given type
-  loadExistingSavedObjects(type) {
-    return this.savedObjectsClient.find({ type, perPage: 1000 });
+  loadExistingSavedObjects(type: string) {
+    return this.savedObjectsClient!.find({ type, perPage: 1000 });
```

Comment:
As mentioned above, might be worth checking that `savedObjectsClient` has been set.